pkg/sql/internal.go

```diff
@@ -837,6 +1006,41 @@ func (ie *InternalExecutor) execInternal(
 	return r, nil
 }
 
+// ReleaseSchemaChangeJobRecords is to release the schema change job records.
+func (ie *InternalExecutor) releaseSchemaChangeJobRecords() {
+	for k := range ie.extraTxnState.schemaChangeJobRecords {
+		delete(ie.extraTxnState.schemaChangeJobRecords, k)
+	}
+}
+
+// CommitTxn is to commit the txn bound to the internal executor.
+func (ie *InternalExecutor) commitTxn(ctx context.Context, txn *kv.Txn) error {
```

Comment:
Can you say more about when this gets called and how the txn relates to, well, `ie.extraTxnState`?

Reply:
It should only be called in `collectionFactory.TxnWithExecutor()`, where we create a new set of schema change job record and schema changer state for the internal executor. 

> how the txn relates to, well, ie.extraTxnState

Yes, it can be confusing. I should have stored the txn from `CollectionFactory` to `ie.extraState.txn` and used it in this function.